programs/config/src/lib.rs

```diff
@@ -78,125 +79,97 @@ mod tests {
-
-        assert_eq!(id(), accounts[1].owner);
+        let (bank, from_keypair) = create_bank(10_000);
+        let config_client = create_config_client(&bank, from_keypair);
+        let config_account = bank.get_account(&config_client.pubkey()).unwrap();
+        assert_eq!(id(), config_account.owner);
         assert_eq!(
             MyConfig::default(),
-            MyConfig::deserialize(&accounts[1].data).unwrap()
+            MyConfig::deserialize(&config_account.data).unwrap()
         );
     }
 
     #[test]
     fn test_process_store_ok() {
         solana_logger::setup();
-        let config_account_keypair = Keypair::new();
-        let config_account = create_config_account();
-
-        let new_config_state = MyConfig::new(42);
+        let (bank, from_keypair) = create_bank(10_000);
+        let config_client = create_config_client(&bank, from_keypair);
+        let config_pubkey = config_client.pubkey();
 
-        let transaction = ConfigTransaction::new_store(
-            &config_account_keypair,
-            &new_config_state,
-            Hash::default(),
-            0,
-        );
-
-        let mut accounts = vec![config_account];
-        process_transaction(&transaction, &mut accounts).unwrap();
+        let my_config = MyConfig::new(42);
+        let instruction = ConfigInstruction::new_store(&config_pubkey, &my_config);
+        config_client.process_instruction(instruction).unwrap();
 
+        let config_account = bank.get_account(&config_pubkey).unwrap();
         assert_eq!(
-            new_config_state,
-            MyConfig::deserialize(&accounts[0].data).unwrap()
+            my_config,
+            MyConfig::deserialize(&config_account.data).unwrap()
         );
     }
 
     #[test]
     fn test_process_store_fail_instruction_data_too_large() {
         solana_logger::setup();
-        let config_account_keypair = Keypair::new();
-        let config_account = create_config_account();
-
-        let new_config_state = MyConfig::new(42);
+        let (bank, from_keypair) = create_bank(10_000);
+        let config_client = create_config_client(&bank, from_keypair);
+        let config_pubkey = config_client.pubkey();
 
-        let mut transaction = ConfigTransaction::new_store(
-            &config_account_keypair,
-            &new_config_state,
-            Hash::default(),
-            0,
-        );
+        let my_config = MyConfig::new(42);
+        let instruction = ConfigInstruction::new_store(&config_pubkey, &my_config);
 
         // Replace instruction data with a vector that's too large
+        let script = Script::new(vec![instruction]);
+        let mut transaction = script.compile();
         transaction.instructions[0].data = vec![0; 123];
-
-        let mut accounts = vec![config_account];
-        process_transaction(&transaction, &mut accounts).unwrap_err();
-    }
-
-    #[test]
-    fn test_process_store_fail_account0_invalid_owner() {
```

Comment:
We can’t represent this test in the new model?

Reply:
Done